flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/DataSetTumbleTimeWindowAggReduceCombineFunction.scala

```diff
@@ -38,19 +40,35 @@ import org.apache.flink.types.Row
   * @param keysAndAggregatesArity The total arity of keys and aggregates
   */
 class DataSetTumbleTimeWindowAggReduceCombineFunction(
-    genAggregations: GeneratedAggregationsFunction,
+    genPreAggregations: GeneratedAggregationsFunction,
+    genFinalAggregations: GeneratedAggregationsFunction,
     windowSize: Long,
     windowStartPos: Option[Int],
     windowEndPos: Option[Int],
     keysAndAggregatesArity: Int)
   extends DataSetTumbleTimeWindowAggReduceGroupFunction(
-    genAggregations,
+    genFinalAggregations,
     windowSize,
     windowStartPos,
     windowEndPos,
     keysAndAggregatesArity)
     with CombineFunction[Row, Row] {
 
+  protected var preAggfunction: GeneratedAggregations = _
```

Comment:
Same as the above, so we can not reuse DataSetTumbleTimeWindowAggReduceGroupFunction